src/controllers/trust.information.controller.ts

```diff
@@ -22,8 +25,80 @@ export const get = (req: Request, res: Response, next: NextFunction) => {
   }
 };
 
-export const post = (req: Request, res: Response) => {
-  logger.debugRequest(req, `POST ${config.TRUST_INFO_PAGE}`);
+export const post = (req: Request, res: Response, next: NextFunction) => {
+  try {
+    logger.debugRequest(req, `POST ${config.TRUST_INFO_PAGE}`);
+
+    // If only one BO is selected, data is a string.
+    // If multiple selected, data is an array.
+    const beneficialOwnerIds: string[] = [];
+    if (typeof req.body.beneficialOwners === 'string') {
+      beneficialOwnerIds.push(req.body.beneficialOwners);
+
+    } else {
+      for (const bo in req.body.beneficialOwners) {
+        beneficialOwnerIds.push(req.body.beneficialOwners[bo]);
+      }
+    }
+
+    const trustData: trustType.Trust[] = JSON.parse(req.body.trusts);
+    const trustsReq: trustType.Trusts = {
+      trusts: trustData
+    };
+
+    // Generate unique trust_id for each trust
+    let trustCount = 0;
+    if (req.session?.data.extra_data.roe.trusts !== undefined) {
+      trustCount = (req.session?.data.extra_data.roe.trusts)?.length;
+    }
+
+    const trustIds: string[] = [];
+    for (const i in trustData) {
+      trustCount++;
+      trustData[i].trust_id = trustCount.toString();
+      trustIds.push(trustCount.toString());
+    }
+
+    assignTrustIdsToBeneficialOwners(req, beneficialOwnerIds, trustIds);
+
+
+    const data: ApplicationDataType = prepareData(trustsReq, TrustKeys);
 
-  return res.redirect("TODO");
+    for (const i in data[TrustKey]) {
+      setApplicationData(req.session, data[TrustKey][i], TrustKey);
+    }
+
+    if (req.body.add) {
+      return res.redirect(config.TRUST_INFO_URL);
+    }
+    if (req.body.submit) {
+      return res.redirect(config.CHECK_YOUR_ANSWERS_PAGE);
+    }
+  } catch (error) {
+    logger.errorRequest(req, error);
+    next(error);
+  }
+};
+
+const assignTrustIdsToBeneficialOwners = (req: any, beneficialOwnerIds: string[], trustIds: string[]) => {
```

Comment:
This function should be moved in the SDK somewhere [here](https://github.com/companieshouse/api-sdk-node/blob/fa6a7ed048635016bc5d20a5b11049075a9b47ad/src/services/overseas-entities/mapping.ts#L18) if we want to map the BOs and trusts IDs only after the user has completed the WEB journey, that's include inserting, updating or deleting. Will simplify the work and will be easier to maintain

Reply:
Agreed not to move to SDK